src/sagas/background-refresh.js

```diff
@@ -0,0 +1,213 @@
+
+  // update counts
+  yield put(Actions.updateVmsPoolsCount())
+}
+
+function* refreshDetailPage ({ id, onNavigation, onSchedule }) {
+  yield selectVmDetail(Actions.selectVmDetail({ vmId: id }))
+  yield getConsoleOptions(Actions.getConsoleOptions({ vmId: id }))
+
+  // Load ISO images on manual refresh click only
+  if (!onNavigation && !onSchedule) {
+    yield fetchIsoFiles(Actions.getIsoFiles())
+  }
+}
+
+function* refreshDialogPage ({ id, onNavigation, onSchedule }) { // TODO: Rename to `refreshCreatePage`
+  if (id) {
+    yield selectVmDetail(Actions.selectVmDetail({ vmId: id }))
+  }
+
+  // Load ISO images on manual refresh click only
+  if (!onNavigation && !onSchedule) {
+    yield fetchIsoFiles(Actions.getIsoFiles())
+  }
+}
+
+function* refreshConsolePage ({ id }) {
+  if (id) {
+    yield selectVmDetail(Actions.selectVmDetail({ vmId: id }))
+  }
+}
+
+function* startSchedulerWithFixedDelay (action) {
+  // if a scheduler is already running, stop it
+  yield put(Actions.stopSchedulerFixedDelay())
+
+  // run a new scheduler
+  yield schedulerWithFixedDelay(action.payload.delayInSeconds)
+}
+
+let _SchedulerCount = 0
+
+function* schedulerWithFixedDelay (delayInSeconds = AppConfiguration.schedulerFixedDelayInSeconds) {
+  if (!isNumber(delayInSeconds) || delayInSeconds <= 0) {
+    return
+  }
+
+  const myId = _SchedulerCount++
+  console.log(`⏰ schedulerWithFixedDelay[${myId}] 🡒 starting fixed delay scheduler`)
+
+  let enabled = true
+  while (enabled) {
+    console.log(`⏰ schedulerWithFixedDelay[${myId}] 🡒 stoppable delay for: ${delayInSeconds}`)
+    const { stopped } = yield race({
+      stopped: take(C.STOP_SCHEDULER_FIXED_DELAY), // TODO: stop the scheduler if an error page or logged out page is displayed
+      fixedDelay: call(delay, (delayInSeconds * 1000)),
+    })
+
+    if (stopped) {
+      enabled = false
+      console.log(`⏰ schedulerWithFixedDelay[${myId}] 🡒 scheduler has been stopped`)
+      continue
+    }
+
+    const isTokenExpired = yield select(state => state.config.get('isTokenExpired'))
+    if (isTokenExpired) {
+      enabled = false
+      console.log(`⏰ schedulerWithFixedDelay[${myId}] 🡒 scheduler has been stopped due to SSO token expiration`)
+      continue
+    }
+
+    const oVirtVersion = yield select(state => state.config.get('oVirtApiVersion'))
+    if (!oVirtVersion.get('passed')) {
+      console.log(`⏰ schedulerWithFixedDelay[${myId}] 🡒 event skipped since oVirt API version does not match`)
+      continue
+    }
+
+    console.log(`⏰ schedulerWithFixedDelay[${myId}] 🡒 running after delay of: ${delayInSeconds}`)
+    yield put(Actions.refresh({
+      onSchedule: true,
+      shallowFetch: true,
+      page: yield select(state => state.vms.get('page')),
+    }))
+  }
+}
+
+export default [
+  takeEvery(C.START_SCHEDULER_FIXED_DELAY, startSchedulerWithFixedDelay),
+  throttle(5000, C.REFRESH_DATA, refreshData),
```

Comment:
This was changed from 1000 to 5000, please add that to patch description as well.

Reply:
Will do.  Really it just help to cut down the number of refreshes that get called if a user gets click happy on the refresh button.